v0/destinations/awin/utils.js

```diff
@@ -0,0 +1,89 @@
+const {
+  isDefinedAndNotNullAndNotEmpty,
+  isDefinedAndNotNull
+} = require("../../util");
+
+let totalAmount;
+const channel = "aw";
+let commissionGroup;
+let voucherCode;
+let currencyCode;
+let orderReference;
+let testMode;
+let cks;
+
+const setParams = message => {
+  const properties = message?.properties;
+  if (isDefinedAndNotNullAndNotEmpty(properties)) {
+    totalAmount =
+      properties.revenue ||
+      properties.amount ||
+      properties.totalAmount ||
+      properties.total_amount;
+    commissionGroup =
+      properties.commissionGroup || properties.commission_group || "DEFAULT";
+    voucherCode = properties.voucherCode || properties.voucher_code;
+    currencyCode =
+      properties.currency ||
+      properties.currencyCode ||
+      properties.currency_code;
+
+    orderReference =
+      properties.order_id ||
+      properties.orderId ||
+      properties.orderReference ||
+      properties.order_reference;
+    testMode =
+      properties.testMode ||
+      properties.test_mode ||
+      properties.isTest ||
+      properties.is_test ||
+      0;
+    cks = properties.cks || properties.awc || "awc";
+  }
```

Comment:
We can use constructPayload function here.

Reply:
Not sure, why we need constructPayload function here.
Do you mean to keep these things in mappingConfig and store them in the payload to retrieve the params object directly(to use it later)?

Reply:
> Do you mean to keep these things in mappingConfig

Yes. Then you can use constructPayload and get the params object and use it directly in generateParams function. This is a better approach the current one (setting global variables and then use them later)

Reply:
Done